vim/merlin/autoload/merlin.py

```diff
@@ -607,12 +622,36 @@ def vim_case_analysis():
                                           "-end", fmtpos(tmp['end']))
         tmp = result[0]
         txt = result[1]
-        replace_buffer_portion(tmp['start'], tmp['end'], txt)
+        replace_buffer_portion(tmp['start'], tmp['end'], txt, next_hole = True)
+
     except MerlinExc as e:
         try_print_error(e)
 
     vim_type_reset()
 
+def vim_previous_hole():
+    line, col = vim.current.window.cursor
+    lst = command_holes()
+    lst.reverse()
+    holes = map(lambda x: x['start'], lst)
+    for pos in holes:
+      hnum = pos['line']
+      hcol = pos['col']
+      if (hnum, hcol) < (line, col):
+        vim.current.window.cursor = (hnum, hcol)
+        break
+
+def vim_next_hole(min = 0, max = sys.maxint):
+    line, col = vim.current.window.cursor
+    lst = command_holes()
+    holes = map(lambda x: x['start'], lst)
+    for pos in holes:
+      hnum = pos['line']
+      hcol = pos['col']
+      if hnum >= min and (hnum, hcol) > (line, col) and hnum <= max:
```

Comment:
Because the list is sorted this could be optimized by breaking if the `max` has been passed. 
But I am not sure it is worth it.